Query parameters for Postgres (Parametrized Queries)

krynble opened the pull request:
This PR addes some minor enhancements to https://github.com/n8n-io/n8n/pull/1471

# Background:

Allow the usage of parameters in SQL queries with postgres. Examples:

`UPDATE orders SET amount = $1, deliveryDate = $2 where id = $3`

Instead of using regular n8n expressions, using parametrized queries add a lot of flexibility without being so verbose.

Backward compatible with existing usages of this node.

This is part of this topic: https://community.n8n.io/t/postgres-node-effective-way-to-upsert-insert-or-update/4315/2

## Enhancements:
Added another parameter that toggles the display of the parameters field so it does not look like something mandatory.

arjones committed:
Adding support to ParameterizedQuery on Postgres Node

krynble committed:
Merge branch 'master' into pr/1471

krynble committed:
Created another parameter to toggle on replacement so it's clear to users what is happening.

krynble committed:
Fixed lint issues

janober committed:
:twisted_rightwards_arrows: Merge branch 'master' into postgres-param-queries

janober committed:
:zap: Formatting

krynble committed:
Improvements to questDB node so it is more consistent

krynble committed:
Merge branch 'postgres-param-queries' of github.com:n8n-io/n8n into postgres-param-queries

krynble committed:
Fixed lint issues

krynble committed:
Fixed typing issue

krynble commented:
Hey @janober This PR contains a few adjustments over #1471 

Had to fix some issues with QuestDB as they were sharing the code but I had to separate it as the database behavior is slightly different. Only part of the protocol is the same.

BHesseldieck commented on packages/nodes-base/nodes/Postgres/Postgres.node.ts:
```diff
@@ -72,10 +72,38 @@ export class Postgres implements INodeType {
 				},
 				default: '',
-				placeholder: 'SELECT id, name FROM product WHERE id < 40',
+				placeholder: 'SELECT id, name FROM product WHERE qty > $1 AND price <= $2',
```
would not abbreviate quantity
```suggestion
				placeholder: 'SELECT id, name FROM product WHERE quantity > $1 AND price <= $2',
```

BHesseldieck commented on packages/nodes-base/nodes/Postgres/Postgres.node.ts:
```diff
@@ -72,10 +72,38 @@ export class Postgres implements INodeType {
+					},
+				},
+				default: '',
+				placeholder: 'qty,price',
```
ref previous comment
```suggestion
				placeholder: 'quantity,price',
```

BHesseldieck commented on packages/nodes-base/nodes/QuestDb/QuestDb.node.ts:
```diff
@@ -73,6 +75,34 @@ export class QuestDb implements INodeType {
+					},
+				},
+				default: '',
+				placeholder: 'qty,price',
```
ref comment in `Postgres.node.ts`

would also update the query placeholder accordingly 👍🏼 

```suggestion
				placeholder: 'quantity,price',
```

BHesseldieck commented on packages/nodes-base/nodes/QuestDb/QuestDb.node.ts:
```diff
@@ -182,22 +228,34 @@ export class QuestDb implements INodeType {
-					if (typeof item.json[col] === 'string') {
-						return `\'${item.json[col]}\'`;
-					} else {
+				const values : string = columnNames.map((col : string) => {
```
```suggestion
				const values: string = columnNames.map((col : string) => {
```

BHesseldieck commented on packages/nodes-base/nodes/QuestDb/QuestDb.node.ts:
```diff
@@ -182,22 +228,34 @@ export class QuestDb implements INodeType {
 			});
 
-			await db.any(pgp.helpers.concat(queries));
+			for(let i = 0; i < queries.length; i++) {
+				// We have to insert lines one by one and wait
+				// Otherwise we might get `table is busy` errors.
+				await db.any(queries[i]);
+			}
```
this might be more readable, but did not test it!
```suggestion
			for(const query of queries) {
				// We have to insert lines one by one and wait
				// Otherwise we might get `table is busy` errors.
				await db.any(query);
			}
```

BHesseldieck reviewed (approved):
awesome job! 🎉 just added some minor comments/suggestions

janober committed:
:zap: Apply suggestions BHesseldieck

Co-authored-by: Ben Hesseldieck <1849459+BHesseldieck@users.noreply.github.com>

krynble committed:
Standardized output for postgres and postgres-like nodes

This changes the behavior of CrateDB, Postgres, QuestDB and TimescaleDB
nodes.

The Execute Query operation used to execute multiple queries but return
the result from only one of the queries.

This change causes the node output to containt results from all queries
that ran, making the behavior more consistent across all n8n.

krynble committed:
Fixing conflicts

krynble committed:
Fixing lint issues

janober committed:
:twisted_rightwards_arrows: Merge branch 'master' into postgres-param-queries

janober committed:
:zap: Minor improvements

janober committed:
:zap: Fix breaking changes files

janober commented:
Thanks a lot. Got merged.

janober commented:
Got released with n8n@0.118.0
